### PR TITLE
added leaky relu activation and tests

### PR DESCRIPTION
**Pull request by psteinb**
I am suggesting to add the leakyReLU activation. For this, I created an autonomous module in `escnn/nn/modules/nonlinearities` and created a test suite adapted from [test/nn/test_fourier_relu.py](https://github.com/QUVA-Lab/escnn/blob/93146fb23c1b31cc130e2aecad8090589219f9f8/test/nn/test_fourier_relu.py). For the latter, I wasn't sure what all the tests and some of the code in "main" was all about, so I just copied it and changed as little as possible.

### CODE REVIEW

**Comment by Gabri95** [comment]
Hi @psteinb 

Thanks for the pull request!

Do you want to use this layer with a continuous group like SO(3) or SO(2) or over a discrete group (like the group of N rotations)?
I am asking this because the LeakyReLU module you implemented assumes a permutation representation like the regular representation (which supports "pointwise" non-linearities) which is only available for finite groups.
However, the test you implemented is based on a discretized Fourier transform which is typically used for continuous groups (this is what I use for FourierPointwise non-linearities).

In other words, are you looking for a module like [ELU](https://quva-lab.github.io/escnn/api/escnn.nn.html#elu) or like [FourierELU](https://quva-lab.github.io/escnn/api/escnn.nn.html#fourierelu)?

Of course, both options are possible! 


Best,
Gabriele

**Comment by psteinb** [comment]
At this point, I am interested in discrete groups for the time being. Once, I learned enough about these, I'd love to tread further into the continuous groups (perhaps in 1-2 months time).

More concretely:  I now need something like [ELU](https://quva-lab.github.io/escnn/api/escnn.nn.html#elu) as I am working with discrete groups. As I figured, others might like the leakyReLU too, I wanted to contribute.
I need the [FourierELU](https://quva-lab.github.io/escnn/api/escnn.nn.html#fourierelu) in 1-2 months time perhaps.

**Comment by Gabri95** [comment]
Alright, all clear!

I will fix this PR then

Thanks,
Gabriele

**Comment by Gabri95** [comment]
For the moment, I only included the Leaky ReLU for discrete groups.

The reason why I didn't  also include the version for continuous groups is that continuous groups rely on a discretized Fourier transform and very "sharp" non-linearities (like ReLU, but I guess also Leaky ReLU) tend to generate more aliasing noise than smoother activations (like ELU).
Implementing Leaky ReLU is not a problem, but I want to encourage the users to use ELU in this case, at least for the moment.


Best,
Gabriele
